.github/workflows/ci.yml

```
@@ -20,24 +24,52 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.17.x
-      - name: Retrieve get module name
-        run: |
-          echo "::set-output name=Module::$(go list -m)"
-        id: module-name
-      - name: Verify builds with xk6
+          go-version: 1.20.x
+      - name: Check build
         run: |
-          go install go.k6.io/xk6/cmd/xk6@latest
-          xk6 build --with ${{ steps.module-name.outputs.Module }}=.
+          go version
+          pwd && ls -l
+          
+          go install go.k6.io/xk6/cmd/xk6@master
+          MODULE_NAME=`go list -m`
+          
+          GOPRIVATE="go.k6.io/k6" xk6 build \
+            --output ./k6ext \
+            --with $MODULE_NAME="."
+          ./k6ext version
 
-  run-unit-tests:
-    runs-on: ubuntu-latest
+  test-go-versions:
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.19.x, 1.20.x, tip]
+        platform: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
-      - name: Install Go
+      - name: Install Go ${{ matrix.go-version }}
+        if: matrix.go-version != 'tip'
         uses: actions/setup-go@v3
         with:
-          go-version: 1.17.x
-      - name: Run unit tests
-        run: go test -v -cover -race ./...
+          go-version: ${{ matrix.go-version }}
+      - name: Install Go stable
+        if: matrix.go-version == 'tip'
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.x
+      - name: Install Go tip
+        shell: bash
+        if: matrix.go-version == 'tip'
+        run: |
+          go install golang.org/dl/gotip@latest
+          gotip download
+          echo "GOROOT=$HOME/sdk/gotip" >> "$GITHUB_ENV"
+          echo "GOPATH=$HOME/go" >> "$GITHUB_ENV"
+          echo "$HOME/go/bin" >> "$GITHUB_PATH"
+          echo "$HOME/sdk/gotip/bin" >> "$GITHUB_PATH"
+      - name: Run tests
+        run: |
+          which go
+          go version
+          go test -race -timeout 800s ./...
```

800s is way too long for just one test :)
```suggestion
          go test -race -timeout 60s ./...
```

---

Copy/paste from extension CI workflows...I'll need to track down the original one as this has been proliferated.